communicator/vmturbocommunicator/server_message_handler.go

```diff
@@ -268,17 +278,45 @@ func (handler *MesosServerMessageHandler) HandleAction(serverMsg *comm.Mediation
 func (handler *MesosServerMessageHandler) NewMesosProbe(previousUseMap map[string]*util.CalculatedUse) (*util.MesosAPIResponse, error) {
 	fullUrl := "http://" + handler.meta.MesosIP + ":" + handler.meta.MesosPort + "/state"
 	glog.V(4).Infof("The full Url is ", fullUrl)
+
 	req, err := http.NewRequest("GET", fullUrl, nil)
 
+	req.Header.Add("content-type", "application/json")
+
+	if handler.meta.DCOS {
+		req.Header.Add("authorization", "token="+handler.meta.Token)
+	}
+
 	glog.V(4).Infof("%+v", req)
 	client := &http.Client{}
 	resp, err := client.Do(req)
+
 	if err != nil {
-		glog.Errorf("Error getting response: %s \n", err)
+		glog.V(3).Infof("Error in GET request to mesos master: %s\n", err)
 		return nil, err
 	}
-	defer resp.Body.Close()
+	// Get token if response if OK
+	if resp.Status == "" {
+		glog.V(3).Infof("Empty response status\n")
+		return nil, err
+	}
+
+	statusmsg := strings.Split(resp.Status, " ")
 
+	if statusmsg[0] != "200" && statusmsg[1] == "Unauthorized" {
```

Comment:
I saw it can be 401, or 403 but we don't know for sure so I will make it != 200.
In the dcos-auth there is a list of options, do you guys think it's worthwhile to do that list now or should we wait until we find out more from getty/when we fix our dcos cluster login ?